File: 2019/Day 01/main.hs
module Main where
import Text.Printf

main :: IO()
main = do
  ys <- total <$> input
  printf "Silver: %d\n"
    (head ys)
  printf "Gold: %d\n"
    (sum $ takeWhile (> 0) ys)

input :: IO [Int]
input = map read.lines
  <$> readFile "input.txt"

total :: [Int] -> [Int]
total = map sum . tail
  . iterate (map fuel)

fuel :: Int -> Int
fuel x = max 0 $
  x `div` 3 - 2

File: 2019/Day 01/main.rb
ys, xs = Array.new, File.
  readlines('input.txt').map(&:to_i)
while xs.any?{|x| x > 0}
  ys << xs.map!{|x| [0,
    x/3 - 2].max}.sum end
puts "Silver: #{ys.first}",
  "Gold: #{ys.sum}"
